locator/token_metadata.cc

```diff
@@ -1261,13 +1259,30 @@ topology::topology(config cfg)
     _pending_locations[utils::fb_utilities::get_broadcast_address()] = std::move(cfg.local_dc_rack);
 }
 
-topology::topology(const topology& other)
-        : _dc_endpoints(other._dc_endpoints)
-        , _dc_racks(other._dc_racks)
-        , _current_locations(other._current_locations)
-        , _pending_locations(other._pending_locations)
-        , _sort_by_proximity(other._sort_by_proximity)
-{
+future<> topology::copy_gently(const topology& other) {
+    co_await clear_gently();
+    _dc_endpoints.reserve(other._dc_endpoints.size());
+    for (const auto& p : other._dc_endpoints) {
+        _dc_endpoints.emplace(p);
+        co_await coroutine::maybe_yield();
```

Comment:
I don't think yielding here is needed. Even a huge cluster will only have a few hundred endpoints. At a very generous 1us/endpoint (several thousand cycles) this is still fast.